fedimint-client/src/db.rs

```diff
@@ -557,3 +557,58 @@ pub async fn remove_old_and_persist_new_inactive_states(
         dbtx.insert_new_entry(&state, &inactive_state).await;
     }
 }
+
+/// Migrates a particular state by looping over all active and inactive states.
+/// If the `migrate` closure returns `None`, this state was not migrated and
+/// should be added to the new state machine vectors.
+pub async fn migrate_state<S: State + IntoDynInstance<DynType = DynState>>(
+    module_instance_id: ModuleInstanceId,
+    active_states: Vec<(Vec<u8>, OperationId)>,
+    inactive_states: Vec<(Vec<u8>, OperationId)>,
+    decoders: ModuleDecoderRegistry,
```

Comment:
Do we need decoders?

Reply:
Decoders are convenient for how we've structured the db migrations right now, but once we switch to returning a `Vec<u8>` they won't be necessary